Source/Core/Core/AchievementManager.cpp

```diff
@@ -506,6 +509,11 @@ void AchievementManager::HandleAchievementTriggeredEvent(const rc_runtime_event_
     return;
   it->second.session_unlock_count++;
   m_queue.EmplaceItem([this, runtime_event] { AwardAchievement(runtime_event->id); });
+  AchievementId game_data_index = it->second.game_data_index;
+  OSD::AddMessage(fmt::format("Unlocked: {} ({})", m_game_data.achievements[game_data_index].title,
+                              m_game_data.achievements[game_data_index].points),
+                  OSD::Duration::VERY_LONG,
+                  (hardcore_mode_enabled) ? OSD::Color::YELLOW : OSD::Color::CYAN);
```

Comment:
Note that this decouples the actual server-side unlocking process from the message displayed to the user. This might be fine, I'm not sure how other emulators handle this, but think about the implications.

Reply:
That is actually deliberate. Some cases (unofficial achievements, Encore Mode) we want to send a message even if we're not actively posting to the site. And a later change will add retry functionality in case the player loses connection to the server - honestly that one's more of a self-serving thing, so I can bring my Deck on flights and the site unlocks everything for me when I land and can turn airplane mode off again.

Reply:
In any case, to my understanding we want to notify the player as soon as the criteria are met, independent of what's going on with respect to the server, but I'll double check.

Reply:
Okay, then I suppose this is fine as-is for now. There probably also should be some kind of 'hey you're about to close Dolphin but there's still outstanding unlocks to RA, if you close Dolphin now you will lose achievement progress' kinda message in the future as well.

Reply:
There will be, haven't quite worked through how that all is going to work yet.